dynamiqs/time_array.py

```diff
@@ -206,13 +206,13 @@ def timecallable(
         >>> f = lambda t: dq.asqarray([[t, 0], [0, 1 - t]])
         >>> H = dq.timecallable(f)
         >>> H(0.5)
-        QArray: shape=(2, 2), dims=(2,), dtype=float32, layout=dense
-        [[0.5 0. ]
-         [0.  0.5]]
+        QArray: shape=(2, 2), dims=(2,), dtype=complex64, layout=dense
+        [[0.5+0.j 0. +0.j]
+         [0. +0.j 0.5+0.j]]
         >>> H(1.0)
-        QArray: shape=(2, 2), dims=(2,), dtype=float32, layout=dense
-        [[1. 0.]
-         [0. 0.]]
+        QArray: shape=(2, 2), dims=(2,), dtype=complex64, layout=dense
+        [[1.+0.j 0.+0.j]
+         [0.+0.j 0.+0.j]]
```

Comment:
This is quite annoying. Is there a simple workaround? (we can merge the PR and fix it later)

Reply:
Also in `dynamiqs/utils/quantum_utils/general.py`.

Reply:
What do you find annoying? I think a QArray should always be complex valued, so IMO it's expected. We don't have to enforce this dtype (easy to change), but then the dtype can be `int32` or whatever else.